src/main/java/controllers/InputParser.java

```diff
@@ -0,0 +1,218 @@
+        } else if (isMarkCommand(cmd)) {
+            try {
+                String[] parts = cmd.split(" ");
+                int index = Integer.parseInt(parts[1]);
+                return new MarkTaskCommand(index);
+            } catch (Exception e) {
+                throw new InvalidInputError("Invalid input for mark!");
+            }
+        } else if (isListCommand(cmd)) {
+            return new ListCommand();
+        } else if (isDeleteCommand(cmd)) {
+            try {
+                String[] parts = cmd.split(" ");
+                int index = Integer.parseInt(parts[1]);
+                return new DeleteTaskCommand(index);
+            } catch (Exception e) {
+                throw new InvalidInputError("Invalid input for delete!");
+            }
+
+        } else if (isFindCommand(cmd)) {
+            try {
+                String[] parts = cmd.split(" ");
+                String taskName = parts[1];
+                return new FindCommand(taskName);
+            } catch (Exception e) {
+                throw new InvalidInputError("Invalid input for delete!");
+            }
+        } else if (isUnmarkCommand(cmd)) {
+            try {
+                String[] parts = cmd.split(" ");
+                int index = Integer.parseInt(parts[1]);
+                return new UnmarkTaskCommand(index);
+            } catch (Exception e) {
+                throw new InvalidInputError("Invalid input for unmark!");
+            }
+        } else if (isTodoCommand(cmd)) {
+            try {
+                Todo newTask = new Todo(cmd.substring(5));
+                return new AddTodoCommand(newTask);
+            } catch (Exception e) {
+                throw new InvalidInputError("Invalid input for todo!");
+            }
+
+        } else if (isDeadlineCommand(cmd)) {
+            try {
+                String description = extractStringBetweenTwoSubStrings(cmd, "deadline", "/by");
+                String by = extractStringFromSubstringToEnd(cmd, "/by");
+                Deadline deadline = new Deadline(description, by);
+                return new AddDeadlineCommand(deadline);
+            } catch (Exception e) {
+                throw new InvalidInputError("Invalid input for deadline!");
+            }
+
+        } else if (isEventCommand(cmd)) {
+            try {
+                String description = extractStringBetweenTwoSubStrings(cmd, "event", "/from");
+                String from = extractStringBetweenTwoSubStrings(cmd, "/from", "/to");
+                String to = extractStringFromSubstringToEnd(cmd, "/to");
+                Event event = new Event(description, from, to);
+                return new AddEventCommand(event);
+            } catch (Exception e) {
+                throw new InvalidInputError("Invalid input for event!");
+            }
+        } else {
+            String errorString = String.format("%s doesn't exist as a command", cmd);
+            throw new InvalidCommandError(errorString);
+        }
+    }
```

Comment:
Maybe a switch statement can be used here? (the `equals` methods have to be implemented anyways)